framework/src/modules/token/cc_commands/cc_transfer.ts

```diff
@@ -49,104 +46,154 @@ export class CCTransferCommand extends BaseCCCommand {
+
+		/* Checking if the token is supported. If not, it throws an error. */
+		if (!supported) {
+			// this.events.get(CcmTransferEvent).error(
+			// 	ctx,
+			// 	{
+			// 		senderAddress,
+			// 		recipientAddress,
+			// 		tokenID,
+			// 		amount,
+			// 		receivingChainID,
+			// 	},
+			// 	TokenEventResult.TOKEN_NOT_SUPPORTED,
+			// );
+
+			this.events.get(CcmTransferEvent).add(
+				ctx,
+				{
+					senderAddress,
+					recipientAddress,
+					tokenID,
+					amount,
+					receivingChainID,
+					result: TokenEventResult.TOKEN_NOT_SUPPORTED,
+				},
+				[senderAddress, recipientAddress, ownChainID],
+			);
+
+			throw new Error(`tokenID ${tokenID.toString('hex')} is not supported`);
+		}
+
+		if (status !== CCM_STATUS_OK) {
+			recipientAddress = senderAddress;
 		}
 
-		// let { recipientAddress } = params;
-		// if (ccm.status !== CCM_STATUS_OK) {
-		// 	recipientAddress = params.senderAddress;
-		// }
-
-		// const canonicalTokenID = await this._tokenMethod.getCanonicalTokenID(
-		// 	methodContext,
-		// 	params.tokenID,
-		// );
-		// const recipientExist = await this._tokenMethod.accountExists(
-		// 	methodContext,
-		// 	params.recipientAddress,
-		// );
-		// let receivedAmount = params.amount;
-		// if (!recipientExist) {
-		// 	const minBalance = this._minBalances.find(mb => mb.tokenID.equals(canonicalTokenID))?.amount;
-		// 	if (!minBalance || minBalance > params.amount) {
-		// 		if (ccm.fee >= MIN_RETURN_FEE * ctx.ccmSize && ccm.status === CCM_STATUS_OK) {
-		// 			await this._interopMethod.error(methodContext, ccm, CCM_STATUS_MIN_BALANCE_NOT_REACHED);
-		// 		}
-		// 		return;
-		// 	}
-		// 	receivedAmount -= minBalance;
-		// 	const [canonicalChainID, canonicalLocalID] = splitTokenID(canonicalTokenID);
-		// 	if (canonicalChainID.equals(CHAIN_ID_ALIAS_NATIVE)) {
-		// 		const supplyStore = this.stores.get(SupplyStore);
-		// 		const supply = await supplyStore.get(methodContext, canonicalLocalID);
-		// 		supply.totalSupply -= minBalance;
-		// 		await supplyStore.set(methodContext, canonicalLocalID, supply);
-		// 	}
-		// }
-
-		// if (tokenChainID.equals(ownChainID)) {
-		// 	const escrowStore = this.stores.get(EscrowStore);
-		// 	const escrowKey = Buffer.concat([ccm.sendingChainID, tokenLocalID]);
-		// 	const escrowData = await escrowStore.get(methodContext, escrowKey);
-
-		// 	escrowData.amount -= params.amount;
-		// 	await escrowStore.set(methodContext, escrowKey, escrowData);
-		// }
-
-		// const userStore = this.stores.get(UserStore);
-		// await userStore.addAvailableBalanceWithCreate(
-		// 	methodContext,
-		// 	recipientAddress,
-		// 	canonicalTokenID,
-		// 	receivedAmount,
-		// );
+		const userStore = this.stores.get(UserStore);
+		const recipientExist = await userStore.has(
+			methodContext,
+			userStore.getKey(recipientAddress, tokenID),
+		);
+
+		if (!recipientExist) {
+			if (fee < FEE_CCM_INIT_USER_STORE) {
+				throw new Error('Insufficient fee to initialize user account.');
+			} else {
```

Comment:
else is not needed here since it's throwing above

Reply:
@shuse2, how does burn happen in this case?

![image](https://user-images.githubusercontent.com/10962699/196167206-c03c5852-de5f-4189-a4f9-dd0f8acb0060.png)

Reply:
wouldn't this be the same?
```
if (recipeintAddress does not exist)
	if (fee < FEE_CCM_INIT_USER_STORE) {
		throw new Error('Insufficient fee to initialize user account.');
	}
	burn()
}
```